app/controllers/vacancies_controller.rb

```diff
@@ -78,4 +80,15 @@ def trigger_dfe_analytics_event(event_type, event_data)
 
     DfE::Analytics::SendEvents.do([event])
   end
+
+  def set_search_coordinates
+    return if do_not_show_distance?
+
+    @search_coordinates = Geocoding.new(form.to_hash[:location]).coordinates
+  end
+
+  def do_not_show_distance?
+    normalised_query = form.to_hash[:location]&.strip&.downcase
+    normalised_query.nil? || LocationQuery::NATIONWIDE_LOCATIONS.include?(normalised_query) || LocationPolygon.contain?(normalised_query)
```

Comment:
we don't want to show distance if the user searches for a nationwide location such as "England" or if they search for a location we have a polygon for. This is because the coordinates Google (or other providers) use for London could be miles away from the location of the school, even if the school is actually in London which could potentially confuse jobseekers.